### DISCUSSION TITLE
@entangle'd value always evaluates to true because a proxy is true…

### DISCUSSION THREAD

**Discussion by jordyvandomselaar** [livewire/livewire#2026] [1. Help]
I have a component like so:

```ts
window.components.ratingBar = function () {
    return {
        givenRating: 0,
        init() {
            const wireId = (this.$refs.ratingBar as HTMLDivElement).getAttribute('wire:id');
            const wire = window.livewire.find(wireId);
            const enabled = (this.$refs.ratingBar as HTMLDivElement).dataset.enabled !== undefined;

            this.givenRating = wire.entangle('givenRating');

            this.$watch('givenRating', (rating: number) => {
                if(!enabled) {
                    return;
                }

                wire.setGivenRating(rating);
            })
        }
    }
};
```
And the div:

```html
<div :class="{'rating-bar': true, 'locked': givenRating}" x-data="components.ratingBar()" x-init="init()" x-ref="ratingBar" @auth data-enabled @endauth></div>
```

The `locked` class is always present because `givenRating` is a proxy which always evaluates truthy. Is there a way to make it not do that?

**Comment by joshhanley** [comment]
Probably because you are entangling it in your x-init function, try using it in your x-data instead and see if that works.

Also there is a this.$wire helper which means you don't need to manually find the wireId and find the component. It will do that for you. Check out `Interacting With Livewire From Alpine: $wire` section on this page https://laravel-livewire.com/docs/2.x/alpine-js

**Reply by jordyvandomselaar** [reply]
Unfortunately, `this.$wire` always seems to result in `undefined` =/

**Reply by joshhanley** [reply]
If that's the case, check the order your scripts are added in your layout file. Livewire scripts should be first before Alpine and any other Libraries, so those helper functions are then available.
